### PR TITLE
Remove the `imagemargin` field

### LINKED ISSUE #4575: Remove imagemargin and floating in Contao 5
### Description

Many years ago we had `margin before` and `margin after` fields for content elements (not exactly sure about that), which we removed because that was theme-related. I think we should finally remove `imagemargin` and `floating`, too, as this is theme-related as well and should be solved otherwise.

The same goes for the layout options (rows and columns) in the gallery element actually, but this might be another topic.

### PR DESCRIPTION
**Pull request by bezin**
Fixes #4575

### CODE REVIEW

File: core-bundle/src/Image/Studio/Figure.php [hunk 1]
```diff
@@ -203,11 +202,10 @@ public function getOptions(): array
...
      *       template's context and directly access the specific data you need.
      *
-     * @param string|array|null $margin              Set margins that will compose the inline CSS for the "margin" key
-     * @param string|null       $floating            Set/determine values for the "float_class" and "addBefore" keys
-     * @param bool              $includeFullMetadata Make all metadata available in the first dimension of the returned data set (key-value pairs)
+     * @param string|null $floating            Set/determine values for the "float_class" and "addBefore" keys
+     * @param bool        $includeFullMetadata Make all metadata available in the first dimension of the returned data set (key-value pairs)
      */
-    public function getLegacyTemplateData(array|string|null $margin = null, string $floating = null, bool $includeFullMetadata = true): array
+    public function getLegacyTemplateData(string $floating = null, bool $includeFullMetadata = true): array
```
Comment (ausi) [inline core-bundle/src/Image/Studio/Figure.php]: As these methods (getLegacyTemplateData and applyLegacyTemplateData) are getting deprecated in the long term anyways, I’d prefer to keep this parameter to keep the method signature the same. With a comment maybe like `Deprecated parameter, has no effect anymore`

@m-vo WDYT?

Comment (m-vo) [reply core-bundle/src/Image/Studio/Figure.php]: Yes, I think so, too. :+1:

Comment (bezin) [reply core-bundle/src/Image/Studio/Figure.php]: Fixed in 349741f

**Comment by bezin** [comment]
We could also remove `Controller::generateMargin` now (after we deprecated it in 4.13), as this is the only usage of this method.

**Review by ausi — DISMISSED** [review]

File: core-bundle/src/Resources/contao/elements/ContentImage.php [hunk 2]
```diff
@@ -68,7 +68,7 @@ protected function compile()
 
 		if (null !== $figure)
 		{
-			$figure->applyLegacyTemplateData($this->Template, $this->imagemargin);
+			$figure->applyLegacyTemplateData($this->Template, null);
```
Comment (m-vo) [inline core-bundle/src/Resources/contao/elements/ContentImage.php]:
```suggestion
			$figure->applyLegacyTemplateData($this->Template);
```

**Comment by m-vo** [comment]
> We could also remove Controller::generateMargin now (after we deprecated it in 4.13), as this is the only usage of this method.

Do you want to create a second PR for 4.13 for the deprecation? IMHO we could remove the method in this PR then altogether.

**Comment by bezin** [comment]
Yes, that is what I thought of, too.

**Comment by bezin** [comment]
Deprecated in #4794 and removed here.

**Review by ausi — DISMISSED** [review]

File: core-bundle/src/Image/Studio/Figure.php [hunk 3]
```diff
@@ -203,7 +202,7 @@ public function getOptions(): array
      *       when using Twig templates! Instead, add this object to your
      *       template's context and directly access the specific data you need.
      *
-     * @param string|array|null $margin              Set margins that will compose the inline CSS for the "margin" key
+     * @param string|array|null $margin              Set margins that will compose the inline CSS for the "margin" key. Deprecated, does not have any effect!
```
Comment (leofeyer) [inline core-bundle/src/Image/Studio/Figure.php]: Why do we keep the parameter at all?

Comment (bezin) [reply core-bundle/src/Image/Studio/Figure.php]: See https://github.com/contao/contao/pull/4788/files/9e54791c5f7a628cd7986e69730c99da139a0c97#r888876972
@ausi @m-vo

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @bezin.

**Pull request merged**